DependencyInjection/CompilerPass/RegisterCompilerPass.php

```diff
@@ -115,8 +113,21 @@ private function registerClass(ContainerBuilder $container, $prefix, $className)
         $definition = $container->register($id, $className);
         $definition->setAutowired(true);
 
-        if ('command' === $prefix) {
+        $this->customizeServiceDefinition($definition);
+    }
+
+    private function customizeServiceDefinition(Definition $definition)
+    {
+        $class = $definition->getClass();
+
+        // Commands
+        if (is_subclass_of($class, Command::class)) {
             $definition->addTag('console.command');
         }
+
+        // Event subscribers
+        if (is_subclass_of($class, EventSubscriberInterface::class)) {
+            $definition->addTag('kernel.event_subscriber');
+        }
```

Comment:
As pointed by @weaverryan it looks like it can be easily generalized. It would be nice to have simple config option for that.